src/viam/sdk/CMakeLists.txt

```diff
@@ -158,6 +158,18 @@ target_link_libraries(viamsdk
   PRIVATE Threads::Threads
 )
 
+# Build api as a system library so we Wall/Werror don't cause unsolvable compilation failures
+target_include_directories(viamapi SYSTEM INTERFACE)
```

Comment:
[q] Should this be done in the api `CMakeLists`?

Reply:
That's a good question. I put it here under the thinking that we're determining how the `sdk` consumes the `api`, so it's a decision _for_ the sdk, if that makes sense. But there are likely subtleties at play here I'm not considering. I think I'd defer to @acmorrow's thinking on this.

Reply:
I don't think this is quite right. It looks like this is applying something to the include directories for `viamapi` itself, rather than altering how `viamsdk` references the includes it uses to refer to `viamapi`?

Up on line 17 (new) we are already doing some magic to put the `viamapi` includes absolutely first on the include path for `viamsdk` so that we can ensure we point into our generated API stuff if that is in use. I think what should be happening is that that line should be updated to make that a `SYSTEM` include.

Reply:
So a funny thing came out of testing. It looks like we don't need anything at all for this. If I remove this line entirely and leave line 17 above untouched, we successfully build on linux and mac, despite the proto files continuing to generate warnings. Warning-generating code in `sdk` still causes a compilation failure, as desired.

I'm pushing up the change without any sort of extra special treatment for `api`. Happy to chat more async or in person if you have concerns about this.

Reply:
Fascinating. I wonder how that comes to be?